Code/Mantid/Framework/DataHandling/test/CheckMantidVersionTest.h

```diff
@@ -0,0 +1,138 @@
+      "  \"tag_name\": \"" + GitHubVersion + "\",\n"
+      "  \"target_commitish\": \"master\",\n"
+      "  \"name\": \"Release version 3.4.0\",\n"
+      "  \"draft\": false,\n"
+      "  \"author\": {\n"
+      "    \"login\": \"peterfpeterson\",\n"
+      "    \"id\": 404003,\n"
+      "    \"avatar_url\": \"https://avatars.githubusercontent.com/u/404003?v=3\",\n"
+      "    \"gravatar_id\": \"\",\n"
+      "    \"url\": \"https://api.github.com/users/peterfpeterson\",\n"
+      "    \"html_url\": \"https://github.com/peterfpeterson\",\n"
+      "    \"followers_url\": \"https://api.github.com/users/peterfpeterson/followers\",\n"
+      "    \"following_url\": \"https://api.github.com/users/peterfpeterson/following{/other_user}\",\n"
+      "    \"gists_url\": \"https://api.github.com/users/peterfpeterson/gists{/gist_id}\",\n"
+      "    \"starred_url\": \"https://api.github.com/users/peterfpeterson/starred{/owner}{/repo}\",\n"
+      "    \"subscriptions_url\": \"https://api.github.com/users/peterfpeterson/subscriptions\",\n"
+      "    \"organizations_url\": \"https://api.github.com/users/peterfpeterson/orgs\",\n"
+      "    \"repos_url\": \"https://api.github.com/users/peterfpeterson/repos\",\n"
+      "    \"events_url\": \"https://api.github.com/users/peterfpeterson/events{/privacy}\",\n"
+      "    \"received_events_url\": \"https://api.github.com/users/peterfpeterson/received_events\",\n"
+      "    \"type\": \"User\",\n"
+      "    \"site_admin\": false\n"
+      "  }";
+
+    return outputString;
+  }
+  virtual std::string getCurrentVersion() const {
+    return CurrentVersion;
+  }
+  
+
+};
+}
+
+class CheckMantidVersionTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static CheckMantidVersionTest *createSuite() { return new CheckMantidVersionTest(); }
+  static void destroySuite( CheckMantidVersionTest *suite ) { delete suite; }
+
+
+  void test_Init()
+  {
+    CheckMantidVersion alg;
+    TS_ASSERT_THROWS_NOTHING( alg.initialize() )
+    TS_ASSERT( alg.isInitialized() )
+  }
+
+  void runTest(const std::string& localVersion, const std::string& gitHubVersion, bool expectedResult)
+  {
+    MockedCheckMantidVersion alg(localVersion,gitHubVersion);
+    TS_ASSERT_THROWS_NOTHING( alg.initialize() )
+
+    TS_ASSERT( alg.isInitialized() );
+    TS_ASSERT_THROWS_NOTHING( alg.execute(); );
+    TS_ASSERT( alg.isExecuted() );
+    
+    std::string currentVersion = alg.PropertyManagerOwner::getProperty("CurrentVersion");
+    std::string mostRecentVersion = alg.PropertyManagerOwner::getProperty("MostRecentVersion");
+    bool isNewVersionAvailable = alg.PropertyManagerOwner::getProperty("IsNewVersionAvailable");
+
+    // Check the results
+    TS_ASSERT_EQUALS(alg.CurrentVersion,currentVersion);
+    TS_ASSERT_EQUALS(alg.CurrentVersion,currentVersion);
```

Comment:
Is this comparison done twice intentionally? Or maybe you meant to compare the github version?